test/internet/test-http2-https-origin-string-correctness.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+// Ref: https://github.com/nodejs/node/issues/39919
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const http2 = require('http2');
+
+function _verifyOriginSet(session, originString) {
+  session.once('remoteSettings', () => {
+    assert.strictEqual(typeof session.originSet, 'object');
+    assert.strictEqual(session.originSet.length, 1);
+    assert.strictEqual(session.originSet[0], originString);
+    session.close();
+  });
+  session.once('error', (error) => {
+    assert.strictEqual(error.code, 'ECONNREFUSED');
+    session.close();
+  });
+}
+
+function withServerName() {
+  const session = http2.connect('https://1.1.1.1', { servername: 'cloudflare-dns.com' });
```

Comment:
This requires an external connection. Is it possible to use localhost? See https://github.com/nodejs/node/pull/39011